src/Workspaces/Remote/Core/AbstractAssetProvider.cs

```diff
@@ -152,6 +149,18 @@ async Task<ImmutableArray<DocumentInfo>> CreateDocumentInfosAsync(DocumentChecks
         // TODO: do we need version?
         return new DocumentInfo(attributes, textLoader, documentServiceProvider: null);
     }
+
+    public AssetHelper<T> GetAssetHelper<T>()
```

Comment:
this pattern helped make the callsites work better with inference, by splitting the type args into the parts that have to be specified (the `T`) and the part that can be inferred (the `TArgs`).